test/UrlParserTest.cpp

```diff
@@ -83,3 +83,33 @@ TEST(UrlParserTest, parseRequestTarget) {
       parseRequestTarget("file://more-than-target"),
       testing::StrEq("Failed to parse URL: \"file://more-than-target\"."));
 }
+
+TEST(UrlParserTest, parseDatasetClauses) {
+  using namespace ad_utility::url_parser;
+  using Iri = ad_utility::triple_component::Iri;
+
+  // Construct the vector from an initializer list without specifying the type.
+  auto IsDatasets = [](const std::vector<DatasetClause>& datasetClauses)
+      -> testing::Matcher<const std::vector<DatasetClause>&> {
+    return testing::ContainerEq(datasetClauses);
+  };
+
+  EXPECT_THAT(parseDatasetClauses({}), IsDatasets({}));
+  EXPECT_THAT(
+      parseDatasetClauses({{"default-graph-uri", {"https://w3.org/1"}}}),
+      IsDatasets({{Iri::fromIriref("<https://w3.org/1>"), false}}));
```

Comment:
Somehwere (probably in TripleComponentTestHelpers or ValueIdHelpers) we have  a lambda that does Iri::fromIriref and thus can be used to shorten this call, s.t. it only becomes iri("<...>").